### PR TITLE
itch: init at version v1.24.0

### PR DESCRIPTION
**Pull request by Anderssorby**
Add setup for itch.io desktop client. From issue #90233

###### Motivation for this change

Attempting to add the itch.io desktop client via the itch-setup utility. See issue #90233.

###### Things done

Tried to base it on other electron app expressions without success. 

Currently it is failing, but I'm hoping someone knows how to fix it.

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ x ] NixOS (fails)
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/games/itch/default.nix [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+, pkg-config
+, makeWrapper
+, makeDesktopItem
+, glib
+}:
+
+stdenv.mkDerivation rec {
+
+  version = "v1.24.0";
+  name = "itch-setup-${version}";
```
Comment (SuperSandro2000) [inline pkgs/games/itch/default.nix:14]:
```suggestion
stdenv.mkDerivation rec {
  pname = "itch-setup";
  version = "1.24.0";
```

File: pkgs/games/itch/default.nix [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+  src = fetchurl {
+    url = "https://github.com/itchio/itch-setup/archive/${version}.tar.gz";
```
Comment (SuperSandro2000) [inline pkgs/games/itch/default.nix:17]: Please use fetchFromGitHub.

File: pkgs/games/itch/default.nix [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    sha256 = "1fwjg5wiyf0rjnwcxabyk05q1k8k6hj53vh1q2xz49y4fwq9sv1i";
+  };
+
+  buildInputs = [
+    makeWrapper nodejs go glib
+  ];
```
Comment (SuperSandro2000) [inline pkgs/games/itch/default.nix:23]:
```suggestion
  buildInputs = [
    nodejs go glib
  ];
```

File: pkgs/games/itch/default.nix [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+  #   extraEntries = ''
+  #     StartupWMClass=element
+  #   '';
+  # };
+
+  meta = with stdenv.lib; {
+    homepage = "https://itch.io";
+    description =
+      "A desktop application that you can download and run games from itch.io with.";
+    license = licenses.mit;
```
Comment (SuperSandro2000) [inline pkgs/games/itch/default.nix:53]: Missing maintainers and platforms.

File: pkgs/games/itch/default.nix [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+  #   desktopName = "Itch";
+  #   genericName = "Itch.io desktop";
+  #   comment = meta.description;
+  #   categories = "Games;Game-client";
+  #   extraEntries = ''
+  #     StartupWMClass=element
+  #   '';
+  # };
+
+  meta = with stdenv.lib; {
```
Comment (SuperSandro2000) [inline pkgs/games/itch/default.nix:49]:
```suggestion
  meta = with lib; {
```

File: pkgs/games/itch/default.nix [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+, nodejs
+, pkg-config
+, makeWrapper
+, makeDesktopItem
+, glib
+}:
+let
+  lib = stdenv.lib;
+in
+stdenv.mkDerivation rec {
```
Comment (SuperSandro2000) [inline pkgs/games/itch/default.nix:13]:
```suggestion

stdenv.mkDerivation rec {
```

File: pkgs/games/itch/default.nix [hunk 7]
```diff
@@ -0,0 +1,60 @@
+{ stdenv
```
Comment (SuperSandro2000) [inline pkgs/games/itch/default.nix:1]:
```suggestion
{ stdenv
, lib
```

File: pkgs/top-level/all-packages.nix [hunk 8]
```diff
@@ -26459,6 +26459,8 @@ in
   # used as base package for iortcw forks
   iortcw_sp = callPackage ../games/iortcw/sp.nix { };
 
+  itch = callPackage ../games/itch/default.nix {  };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:26462]:
```suggestion
  itch = callPackage ../games/itch/default.nix { };
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by wackbyte** [comment]
any updates on this?

**Comment by davidak** [comment]
This PR is superseded by https://github.com/NixOS/nixpkgs/pull/157921.

**Pull request closed without merging**